### PR TITLE
add batch support for metadata usedBy resource

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 1]
```diff
@@ -308,6 +310,34 @@ public void removeObjByUri(String uri) {
...
+     * Find all objects which use the given objects. Batch alternative to {@link #usedBy(Project, String, boolean, Class[])}
+     * @param project project
+     * @param uris    URIs of object to find using objects for
+     * @param nearest find nearest objects only
+     * @param types   what types (categories) to search for (for example 'reportDefinition', 'report', 'tableDataLoad',
+     *                'table'...)
+     * @return map [object URI] -> [objects using given object]
+     * @see #usedBy(Project, String, boolean, Class[])
+     */
+    public Collection<Usage> usedBy(Project project, Set<String> uris, boolean nearest, Class<? extends Obj>... types) {
```
Comment (martiner) [inline src/main/java/com/gooddata/md/MetadataService.java]: replace Set with COllection

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 2]
```diff
@@ -308,6 +310,34 @@ public void removeObjByUri(String uri) {
     }
 
     /**
+     * Find all objects which use the given objects. Batch alternative to {@link #usedBy(Project, String, boolean, Class[])}
+     * @param project project
+     * @param uris    URIs of object to find using objects for
+     * @param nearest find nearest objects only
+     * @param types   what types (categories) to search for (for example 'reportDefinition', 'report', 'tableDataLoad',
+     *                'table'...)
+     * @return map [object URI] -> [objects using given object]
```
Comment (martiner) [inline src/main/java/com/gooddata/md/MetadataService.java]: different return type

File: src/main/java/com/gooddata/md/MetadataService.java [hunk 3]
```diff
@@ -308,6 +310,34 @@ public void removeObjByUri(String uri) {
     }
 
     /**
+     * Find all objects which use the given objects. Batch alternative to {@link #usedBy(Project, String, boolean, Class[])}
+     * @param project project
+     * @param uris    URIs of object to find using objects for
+     * @param nearest find nearest objects only
+     * @param types   what types (categories) to search for (for example 'reportDefinition', 'report', 'tableDataLoad',
```
Comment (martiner) [inline src/main/java/com/gooddata/md/MetadataService.java]: document what happens when no type is given

File: src/test/java/com/gooddata/ShowcaseAT.java [hunk 4]
```diff
@@ -245,6 +245,22 @@ public void createModel() throws Exception {
...
+
+        metadataService.usedBy(project, uri, false);
+    }
+
+    @Test(groups = "md", dependsOnMethods = "createModel")
+    public void usedByBatch() {
+        final MetadataService metadataService = gd.getMetadataService();
+        final Collection<String> uris = metadataService.findUris(project, identifier("attr.person.department"), identifier("attr.person.role"));
+
+        assertThat(metadataService.usedBy(project, new HashSet<>(uris), false).size(), is(2));
```
Comment (martiner) [inline src/test/java/com/gooddata/ShowcaseAT.java]: hasSize()

File: src/test/java/com/gooddata/ShowcaseAT.java [hunk 5]
```diff
@@ -245,6 +245,22 @@ public void createModel() throws Exception {
         modelService.updateProjectModel(project, projectModelDiff).get();
     }
 
+    @Test(groups = "md", dependsOnMethods = "createModel")
+    public void usedBy() {
+        final MetadataService metadataService = gd.getMetadataService();
+        final String uri = metadataService.findUris(project, identifier("attr.person.department")).iterator().next();
+
+        metadataService.usedBy(project, uri, false);
```
Comment (martiner) [inline src/test/java/com/gooddata/ShowcaseAT.java]: add assert

File: src/test/java/com/gooddata/md/MetadataServiceIT.java [hunk 6]
```diff
@@ -66,6 +71,39 @@ public void testUsedBy() throws Exception {
...
+                .havingMethodEqualTo("POST")
+                .havingPathEqualTo(USEDBY_URI)
+                .respond()
+                .withStatus(200)
+                .withBody(MAPPER.writeValueAsString(useMany));
+
+        final Collection<Usage> result = gd.getMetadataService().usedBy(project, new HashSet<>(asList(OBJ_URI, OBJ_URI2)), false, ReportDefinition.class);
+
+        assertThat(result, is(notNullValue()));
+        assertThat(result.size(), is(2));
```
Comment (martiner) [inline src/test/java/com/gooddata/md/MetadataServiceIT.java]: hasSize()

File: src/main/java/com/gooddata/md/MetadataService.java (hunk 1 above)
Comment (sadam21) [inline src/main/java/com/gooddata/md/MetadataService.java]: Replaced...

File: src/main/java/com/gooddata/md/MetadataService.java (hunk 2 above)
Comment (sadam21) [inline src/main/java/com/gooddata/md/MetadataService.java]: Javadoc fixed...

File: src/main/java/com/gooddata/md/MetadataService.java (hunk 3 above)
Comment (sadam21) [inline src/main/java/com/gooddata/md/MetadataService.java]: Javadoc added...

File: src/test/java/com/gooddata/ShowcaseAT.java (hunk 4 above)
Comment (sadam21) [inline src/test/java/com/gooddata/ShowcaseAT.java]: Used...

File: src/test/java/com/gooddata/ShowcaseAT.java (hunk 5 above)
Comment (sadam21) [inline src/test/java/com/gooddata/ShowcaseAT.java]: Added...

File: src/test/java/com/gooddata/md/MetadataServiceIT.java (hunk 6 above)
Comment (sadam21) [inline src/test/java/com/gooddata/md/MetadataServiceIT.java]: Done.

**Pull request merged**